src/vos/vos_aggregate.c

```diff
@@ -1540,32 +1586,52 @@ join_merge_window(daos_handle_t ih, struct agg_merge_window *mw,
 	recx2ext(&entry->ie_orig_recx, &phy_ext);
 	D_ASSERT(ext1_covers_ext2(&phy_ext, &lgc_ext));
 
+	switch (entry->ie_dtx_state) {
+	case DTX_ST_COMMITTED:
+		break;
+	case DTX_ST_ABORTED:
+		/*
+		 * Delete the aborted entry, and inform iterator to abort
+		 * current evtree aggregation.
+		 *
+		 * NB. We can't continue current evtree aggregation since
+		 * other entry's visibility could be invalid after deleting.
```

Comment:
I do not understand why removing an aborted entry will affect others visibility. The removed one is invisible to others before it is removed since related DTX is 'aborted'.

Reply:
'visibility' here in evtree is about epoch order (extent with higher epoch covers extent with lower epoch, then the extent with lower epoch is invisible), it's not about dtx 'availability'.

Reply:
why does it affect visibilty though?  If it's aborted, can't the evtree sort algorithm just mark it as such and sort it properly?

Reply:
Yes, I think we may optimize it by excluding the aborted entries while sorting. (Simply mark the whole aborted entry as covered or some new flag?).  Could you help on this when you have spare cycle later (I think you know the sorting code much better than me)? Thanks.